src/vivarium_public_health/metrics/mortality.py

```diff
@@ -7,121 +7,173 @@
+
+    # noinspection PyMethodMayBeStatic
+    def _get_causes_of_death(self, builder: Builder) -> List[str]:
+        # todo can we specify only causes with excess mortality?
         diseases = builder.components.get_components_by_type(
             (DiseaseState, RiskAttributableDisease)
         )
-        self.causes = [c.state_id for c in diseases] + ["other_causes"]
-
-        life_expectancy_data = builder.data.load(
-            "population.theoretical_minimum_risk_life_expectancy"
-        )
-        self.life_expectancy = builder.lookup.build_table(
-            life_expectancy_data, key_columns=[], parameter_columns=["age"]
-        )
+        return [c.state_id for c in diseases] + ["other_causes"]
 
+    # noinspection PyMethodMayBeStatic
+    def _get_population_view(self, builder: Builder) -> PopulationView:
         columns_required = [
             "tracked",
             "alive",
-            "entrance_time",
-            "exit_time",
-            "cause_of_death",
             "years_of_life_lost",
-            "age",
+            "cause_of_death",
+            "exit_time",
         ]
-        if self.config.by_sex:
-            columns_required += ["sex"]
-        self.population_view = builder.population.get_view(columns_required)
-
-        builder.value.register_value_modifier("metrics", self.metrics)
+        return builder.population.get_view(columns_required)
 
-    def metrics(self, index, metrics):
-        pop = self.population_view.get(index)
-        pop.loc[pop.exit_time.isnull(), "exit_time"] = self.clock()
+    def _register_collect_metrics_listener(self, builder: Builder) -> None:
+        builder.event.register_listener("time_step", self.on_collect_metrics)
 
-        person_time = get_person_time(
-            pop, self.config.to_dict(), self.start_time, self.clock(), self.age_bins
-        )
-        deaths = get_deaths(
-            pop,
-            self.config.to_dict(),
-            self.start_time,
-            self.clock(),
-            self.age_bins,
-            self.causes,
-        )
-        ylls = get_years_of_life_lost(
-            pop,
-            self.config.to_dict(),
-            self.start_time,
-            self.clock(),
-            self.age_bins,
-            self.life_expectancy,
-            self.causes,
+    def _register_metrics_modifier(self, builder: Builder) -> None:
+        builder.value.register_value_modifier(
+            self.metrics_pipeline_name,
+            modifier=self.metrics,
+            requires_columns=["age", "exit_time", "alive"],
         )
 
-        metrics.update(person_time)
-        metrics.update(deaths)
-        metrics.update(ylls)
+    ########################
+    # Event-driven methods #
+    ########################
+
+    def on_collect_metrics(self, event: Event) -> None:
+        pop = self.population_view.get(event.index)
+        pop_died = pop[(pop["alive"] == "dead") & (pop["exit_time"] > event.time - self.step_size)]
```

Comment:
I think doing exact equality `pop["exit_time"] == event.time` is fine here.  I can't remember right now why we said something different.

Reply:
If you do need the `step_size` it's available as an attribute on the `event`.